### PR TITLE
Fix docstrings and _tags of Detrender, ConditionalDeseasonalizer

### PR DESCRIPTION
**Pull request by aiwalter**
#### What does this implement/fix? Explain your changes.
Fix docstrings and _tags of `Detrender`, `ConditionalDeseasonalizer`

### CODE REVIEW

File: sktime/transformations/series/detrend/_detrend.py [hunk 1]
```diff
@@ -28,26 +28,17 @@ class Detrender(_SeriesToSeriesTransformer):
...
-
-    For example, to remove the linear trend of a time series:
-    forecaster = PolynomialTrendForecaster(degree=1)
-    transformer = Detrender(forecaster=forecaster)
-    yt = transformer.fit_transform(y_train)
-
     Parameters
     ----------
-    forecaster : estimator object, default=None
+    forecaster : sktime-like forecaster, default=None
```
Comment (fkiraly) [inline sktime/transformations/series/detrend/_detrend.py:33]: I would simply say `forecaster : sktime forecaster, follows BaseForecaster template, default = linear trend, i.e., PolynomialTrendForecaster(degree=1)`

Comment (fkiraly) [reply sktime/transformations/series/detrend/_detrend.py:33]: @aiwalter, minor comment on docstring - would you mind changing, then we can get this merged?

**Review by fkiraly — APPROVED** [review]
Looks good to me.

**Review by chrisholder — APPROVED** [review]

**Comment by fkiraly** [comment]
I've integrated this in #1795 directly. The changes kept are those to the main docstrings and the `ConditionalDeseasonlizer` example. Changes to the private docstrings are superseded by consisten changes in the refactor.

**Comment by fkiraly** [comment]
#1795 got merged, with the proposed changes from this PR integrated (see above), hence closing.

@aiwalter, I´ve copied it across almost verbatin - kindly have a look if it´s fine.
Private docstrings have been changed for consistency.

**Pull request closed without merging**